src/components/Carousel.js

```diff
@@ -189,6 +204,76 @@ class Carousel extends Component {
     return props[propName];
   };
 
+  /**
+   * Function that stores indexes in the state of unique slides that should be lazy loaded.
+   */
+  setLazyLoadedSlides() {
+    const children = this.getChildren();
+    const currentSlideIndex = this.getCurrentSlideIndex();
+    const slidesPerScroll = this.getProp('slidesPerScroll');
+    const slidesPerPage = this.getProp('slidesPerPage');
+    const infinite = this.getProp('infinite');
+
+    const prevStep = currentSlideIndex - slidesPerScroll;
+    const nextStep = currentSlideIndex + (slidesPerPage - 1) + slidesPerScroll;
+
+    let hooped = false;
+    let prevIndex = this.clamp(prevStep);
+    let nextIndex = this.clamp(nextStep);
+
+    if (infinite) {
+      const {
+        value: prevValue,
+        hooped: prevHooped,
+      } = this.hoop(prevStep);
+
+      const {
+        value: nextValue,
+        hooped: nextHooped,
+      } = this.hoop(nextStep);
+
+      prevIndex = prevValue;
+      nextIndex = nextValue;
+      hooped = prevHooped || nextHooped;
+    }
+
+    const lazyLoadedSlides = children.map((_, index) => {
+      if (hooped && ((index >= prevIndex) || (index <= nextIndex))) {
```

Comment:
@ndeviant 

could you explain this condition?

if `prevIndex` is less than the `nextIndex`, `((index >= prevIndex) || (index <= nextIndex))` looks to be always satisfied

Reply:
@piotr-s-brainhub for example we have 5 slides, and we're starting from 0 index in infinite mode. We're "hooping" the values, so that prevIndex is now 4(last index) and nextIndex is 1. So we can preload the previous and next slides.
So, if one of indexes is out of range (prevIndex = 0 - 1), we say its "hooped" (prevIndex = 4), and now we should check that all slides from prevIndex till the end of array are loaded. The same approach for the nextIndex.
A bit more complicated example: 
9 slides, `infinite`, `slidesPerScroll=2`, `slidesPerPage=2`, initial index is 7 (so that 7 and 8 indexes are shown initially). 
Prev index will be 5 (5-6 shown) and next index is 1 (0-1). Those are also hooped, cause after 7 and 8 indexes we're came to the beginning of array. Therefore we preload every index from 5 till end of array `(index >= prevIndex)`, and every from 0 till 1 `(index <= nextIndex)`.
I hope that I was able to express myself at least to some extent. :)

Reply:
thanks